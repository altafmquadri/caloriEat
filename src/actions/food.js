/**
 * @function getFood action generator to the food reducer
 * @param {object} food object
 * @returns {object} food object searched for  
 */
export const getFood = (food) => ({
    type: 'GET_FOOD',
    payload: food
})

/**
 * @function getServing action to be dispatched by reducer
 * @param {object} serving food item retrieved from the nutrient API
 * @param {object} objToState add in serving parameters obj to the rest of state
 * @returns {object} the serving measurement of food item
 */
export const getServing = (serving, objToState) => ({
    type: 'GET_SERVING',
    payload: serving,
    obj: objToState
})

/**
 * @function addFood adds the action to add properties of food to the reducer
 * @param {object} food adds food properties of date and mealCategory
 * @returns {object} food with properties to add to state in the reducer
 */
export const addFood = (food) => ({
    type: 'ADD_FOOD',
    payload: food
})

/**
 * @function clearFood dispatches action to clear the food state
 */
export const clearFood = () => ({
    type: 'CLEAR_FOOD'
})

/**
 * @function fetchFood fetch to retrieve food
 * @param {string} lookup to be passed into fetch
 * @returns {function} which will dispatch the action to bring the food item into state
 */
export const fetchFood = (lookup) => {
    return dispatch => {
        return fetch(`https://api.edamam.com/api/food-database/v2/parser?ingr=${lookup}&app_id=${process.env.EDAMAM_APP_ID}&app_key=${process.env.EDAMAM_APP_KEY}`)
            .then(res => res.json())
            .then(food => dispatch(getFood(food)))
    }
}

/**
 * @function fetchServing fetch to retrieve serving
 * @param {object} obj object to be passed into fetch
 * @returns {function} which will dispatch the action to set the new selected serving state
 */
export const fetchServing = (obj) => {
    return dispatch => {
        return fetch(`https://api.edamam.com/api/food-database/v2/nutrients?app_id=${process.env.EDAMAM_APP_ID}&app_key=${process.env.EDAMAM_APP_KEY}`, {
            method: 'POST',
            headers: {
                "Content-type": 'application/json',
                Accepts: 'application/json'
            },
            body: JSON.stringify({
                ingredients: [
                    obj
                ]
            })
        })
            .then(res => res.json())
            .then(serving => dispatch(getServing(serving, obj)))
    }
}

