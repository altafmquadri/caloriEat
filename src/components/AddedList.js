import React from 'react';
import { connect } from 'react-redux'

const AddedList = (props) => {
    console.log(props);
    return (
        <div>
            {props.list.map(meal => (
                <li key={meal.uuid}>{`${meal.foodItem} added, calories:${meal.calories}`}</li>
            ))}
        </div>
    );
}

const mapStateToProps = (state) => ({
    list: state.list
})

export default connect(mapStateToProps)(AddedList)