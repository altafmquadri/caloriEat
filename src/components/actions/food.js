export const getFood = (food) => ({
    type: 'GET_FOOD',
    food
})

export const fetchFood = (lookup) => {
    return dispatch => {
        return fetch(`https://api.edamam.com/api/food-database/v2/parser?ingr=${lookup}&app_id=${process.env.EDAMAM_APP_ID}&app_key=${process.env.EDAMAM_APP_KEY}`)
            .then(res => res.json())
            .then(food => dispatch(getFood(food)))
    }
}