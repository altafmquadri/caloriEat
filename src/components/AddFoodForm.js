import React, { Component } from 'react';
import { connect } from 'react-redux'
import moment from 'moment'
import 'react-dates/initialize'
import 'react-dates/lib/css/_datepicker.css'
import { SingleDatePicker } from 'react-dates'
import { addFood } from './actions/food'
import { addMeal } from './actions/meal'

class AddFoodForm extends Component {
    state = {
        date: moment(),
        mealCategory: 'breakfast',
        calendarFocused: false,
    }

    onDateChange = date => {
        if (date) this.setState({ date })
    }

    onSubmit = e => {
        e.preventDefault()
        const food = {
            date: this.state.date,
            mealCategory: this.state.mealCategory
        }
        this.props.addFood(food) //from mapDispatch to props updates the food state, viewing in redux tools shows proper state for food
        const meal = this.props.food //comes from mapStateToProps 
        console.log(this.props, 'logging props before meal assignment');
        console.log(meal, 'in form') //logging this shows it's not the updated state I am expecting
        this.props.addMeal(meal) //trying to add state from food reducer to the meal reducer, meal reducer is not updated via redux tools
    }

    onFocusChange = ({ focused }) => this.setState({ calendarFocused: focused })
    onMealChange = e => this.setState({ mealCategory: e.target.value })

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <select value={this.state.mealCategory}
                        onChange={this.onMealChange}>
                        <option value='breakfast'>Breakfast</option>
                        <option value='lunch'>Lunch</option>
                        <option value='dinner'>Dinner</option>
                        <option value='snack'>Snack</option>
                    </select>
                    <SingleDatePicker
                        date={this.state.date}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        id="caloriEat-addFood-form" />
                    <button type="submit">Add Food</button>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    addFood: food => dispatch(addFood(food)),
    addMeal: meal => dispatch(addMeal(meal))
})

const mapStateToProps = state => ({
    food: state.food
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFoodForm)