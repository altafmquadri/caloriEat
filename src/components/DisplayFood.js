import React, { Component } from 'react';
import { connect } from 'react-redux'
import { fetchServing } from './actions/food'


class DisplayFood extends Component {
    state = {
        quantity: '',
        measureURI: '',
        foodId: ''
    }

    onChange = e => {
        this.setState({
            quantity: 1,
            measureURI: e.target.value,
            foodId: this.props.food.foodId
        }, () => this.props.fetchServing(this.state))
    }

    render() {
        console.log(this.props);
        const displayContents = () => {
            if (this.props.food.foodItem) {
                const { foodItem, calories, carbohydrates, fat, protein, fiber, image, measures } = this.props.food
                const contents = (
                    <div>
                        <h1>{foodItem}</h1>
                        <h3>Calories: {calories}</h3>
                        <h3>Carbohydrates: {carbohydrates}g</h3>
                        <h3>Fat: {fat}g</h3>
                        <h3>Protein: {protein}g</h3>
                        <h3>Fiber: {fiber}g</h3>
                        <img alt={foodItem} src={image}></img>
                        <select
                            onChange={this.onChange}
                            value={this.state.measureURI}>
                            {measures.map((measure, i) => <option key={i} value={measure.uri}>{measure.label}</option>)}
                        </select>
                    </div >
                )
                return contents
            } else return null

        }
        console.log(this.props);
        return (
            <div>
                {displayContents()}
            </div>

        )
    }
}




const mapStateToProps = (state) => ({
    food: state.food
})

const mapPropsToDispatch = (dispatch) => ({
    fetchServing: obj => dispatch(fetchServing(obj))
})

export default connect(mapStateToProps, mapPropsToDispatch)(DisplayFood)