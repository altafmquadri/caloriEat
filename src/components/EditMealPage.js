import React, { Component } from 'react';
import { connect } from 'react-redux'
import { editServing } from './actions/meal';

class EditMealPage extends Component {
    constructor(props) {
        super(props)
        this.state = {
            uuid: this.props.meal.uuid,
            foodId: this.props.meal.foodId,
            measureURI: this.props.meal.measureURI,
            quantity: this.props.meal.quantity,
            mealCategory: this.props.meal.mealCategory
        }
    }

    onMeasureChange = e => (this.setState({ measureURI: e.target.value }))
    onQuantityChange = e => (this.setState({ quantity: e.target.value }))
    onMealCategoryChange = e => (this.setState({ mealCategory: e.target.value }))

    onSubmit = e => {
        e.preventDefault()
        const { foodId, measureURI, quantity, uuid, mealCategory } = this.state
        const editsObj = {
            foodId,
            measureURI,
            quantity: parseInt(quantity)
        }
        this.props.editServing(uuid, editsObj, mealCategory)

    }

    render() {
        const { foodItem, calories, measures } = this.props.meal
        const { measureURI, quantity, mealCategory } = this.state
        return (
            <div>
                <h1>Edit Meal</h1>
                <h3>{foodItem}</h3>
                <h3>{calories}</h3>
                <div>
                    <form onSubmit={this.onSubmit}>
                        <select
                            value={measureURI}
                            onChange={this.onMeasureChange}>
                            {measures.map((measure, i) => <option key={i} value={measure.uri}>{measure.label}</option>)}
                        </select>
                        <input
                            value={quantity}
                            onChange={this.onQuantityChange}
                            type="text"></input>
                        <select value={mealCategory}
                            onChange={this.onMealCategoryChange}>
                            <option value='breakfast'>Breakfast</option>
                            <option value='lunch'>Lunch</option>
                            <option value='dinner'>Dinner</option>
                            <option value='snack'>Snack</option>
                        </select>
                        <button type="submit">Edit Meal</button>
                    </form>
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    meal: state.meals.find(meal => meal.uuid === props.match.params.uuid)
})

const mapDispatchToProps = dispatch => ({
    editServing: (uuid, fetchObj, mealCategory) => dispatch(editServing(uuid, fetchObj, mealCategory))
})
export default connect(mapStateToProps, mapDispatchToProps)(EditMealPage)