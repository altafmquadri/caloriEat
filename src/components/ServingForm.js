import React, { Component } from 'react';
import { connect } from 'react-redux'
import { fetchServing } from '../actions/food'

class ServingForm extends Component {
    state = {
        quantity: 1,
        measureURI: '',
        foodId: ''
    }

    onSubmit = e => {
        e.preventDefault()
        if (!this.state.measureURI) return
        this.setState({ foodId: this.props.food.foodId }, () => this.props.fetchServing({
            quantity: parseInt(this.state.quantity),
            measureURI: this.state.measureURI,
            foodId: this.state.foodId
        }))
    }

    onChangeServing = (e) => this.setState({ measureURI: e.target.value })
    onChangeQuantity = (e) => this.setState({ quantity: e.target.value })

    render() {
        const { measures } = this.props.food
        return (
            <form onSubmit={this.onSubmit}>
                <select
                    onChange={this.onChangeServing}
                    value={this.state.measureURI}>
                    {measures.map((measure, i) => <option key={i} value={measure.uri}>{measure.label}</option>)}
                </select>

                <input value={this.state.quantity}
                    type="text"
                    onChange={this.onChangeQuantity}>
                </input>

                <button type="submit">Submit</button>
            </form>
        );
    }
}

const mapStateToProps = (state) => ({
    food: state.food
})

const mapPropsToDispatch = (dispatch) => ({
    fetchServing: obj => dispatch(fetchServing(obj))
})

export default connect(mapStateToProps, mapPropsToDispatch)(ServingForm)