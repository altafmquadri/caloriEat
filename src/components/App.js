import React, { Component } from 'react';
import InputForm from './InputForm'
import DisplayFood from './DisplayFood'

class App extends Component {
    render() {
        return (
            <div>
                <InputForm />
                <DisplayFood />
            </div>
        )
    }
}

export default App;