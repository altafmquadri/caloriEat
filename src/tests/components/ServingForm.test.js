import React from 'react';
import { shallow } from 'enzyme'
import { ServingForm } from '../../components/ServingForm'
import { food } from '../fixtures/food'

describe('<ServingForm />', () => {
    test('should render ServingForm', () => {
        const wrapper = shallow(<ServingForm food={food} />)
        expect(wrapper).toMatchSnapshot()
    })
    test('should set Serving on serving change', () => {
        const wrapper = shallow(<ServingForm food={food} />)
        const value = 'cup'
        wrapper.find('select').simulate('change', {
            target: { value }
        })
        expect(wrapper.state('measureURI')).toBe(value)
    })
    test('should set quantity on quantity change', () => {
        const wrapper = shallow(<ServingForm food={food} />)
        const value = '3'
        wrapper.find('input').simulate('change', {
            target: { value }
        })
        expect(wrapper.state('quantity')).toBe(value)
    })
    test('should test onSubmit prop for valid form submission', () => {
        const onSubmit = jest.fn()
        const fetchServing = jest.fn()
        const arg = {
            quantity: expect.any(Number),
            measureURI: expect.any(String),
            foodId: expect.any(Number)
        }
        const wrapper = shallow(<ServingForm
            food={food}
            onSubmit={onSubmit}
            fetchServing={fetchServing} />)
        wrapper.find('form').simulate('submit', {
            preventDefault: () => { }
        })
        expect(fetchServing).toHaveBeenCalled()
    })

})