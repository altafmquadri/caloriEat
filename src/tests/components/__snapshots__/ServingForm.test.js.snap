// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServingForm /> should render ServingForm 1`] = `
<form
  onSubmit={[Function]}
>
  <select
    onChange={[Function]}
    value=""
  >
    <option
      key="0"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_serving"
    >
      Serving
    </option>
    <option
      key="1"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_gram"
    >
      Gram
    </option>
    <option
      key="2"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_ounce"
    >
      Ounce
    </option>
    <option
      key="3"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_pound"
    >
      Pound
    </option>
    <option
      key="4"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_kilogram"
    >
      Kilogram
    </option>
    <option
      key="5"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_tablespoon"
    >
      Tablespoon
    </option>
    <option
      key="6"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_cup"
    >
      Cup
    </option>
    <option
      key="7"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_bottle"
    >
      Bottle
    </option>
    <option
      key="8"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_fluid_ounce"
    >
      Fluid ounce
    </option>
    <option
      key="9"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_quart"
    >
      Quart
    </option>
    <option
      key="10"
      value="http://www.edamam.com/ontologies/edamam.owl#Measure_gallon"
    >
      Gallon
    </option>
  </select>
  <input
    onChange={[Function]}
    type="text"
    value={1}
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
